Libs/Optimize/ParticleSystem/Constraints.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "Constraint.h"
+#include "PlaneConstraint.h"
+#include "SphereConstraint.h"
+#include "FreeFormConstraint.h"
+#include <vector>
+#include "vnl/vnl_inverse.h"
+#include "vnl/vnl_cross.h"
+#include "Eigen/Dense"
+#include "itkPoint.h"
+
+namespace itk
+{
+
+class Constraints{
+public:
+    Constraints(){
+        planeConsts = new std::vector<PlaneConstraint>();
+        sphereConsts = new std::vector<SphereConstraint>();
+        freeFormConsts = new std::vector<FreeFormConstraint>();
+    }
+
+    ~Constraints(){
+        delete planeConsts;
+        delete sphereConsts;
+        delete freeFormConsts;
+    }
+
+  // Set constraints
+  void addPlane(const vnl_vector<double> &a, const vnl_vector<double> &b,const vnl_vector<double> &c);
+  void addSphere(const vnl_vector_fixed<double, DIMENSION> &v, double r);
+  void setFreeFormConstraint(std::string filename);
+
+  // Transforms
+  bool transformConstraints(const vnl_matrix_fixed<double, 4, 4> &Trans);
+  bool transformPlanes(const vnl_matrix_fixed<double, 4, 4> &Trans);
+
+  // Apply functions
+  std::stringstream applyBoundaryConstraints(vnl_vector_fixed<double, 3> &gradE, const Point<double, 3> &pos);
```

Comment:
These apply constraints calls return stringstreams?

Reply:
This is for debugging purposes. Regular couts don't work well with OpenMP optimization. It will get removed at some point.